Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -119,18 +133,45 @@ public bool TryGetXComponentCorrelationId(string instrumentationKey, out string
                             {
                                 this.GenerateCorrelationIdAndAddToDictionary(instrumentationKey, appId.Result);
                             }
+                            catch (AggregateException ex)
+                            {
+                                ex.Flatten();
+                                if (ex.InnerExceptions != null && ex.InnerExceptions.Count > 0 && ex.InnerExceptions[0] != null)
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex.InnerExceptions[0]);
+                                }
+                                else
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex);
+                                }
+                            }
                             catch (Exception ex)
                             {
-                                CrossComponentCorrelationEventSource.Log.FetchAppIdFailed(this.GetExceptionDetailString(ex));
+                                this.RegisterFailure(instrumentationKey, ex);
```

Comment:
This looks like faulty merge. Note that I changed appId.Result on line 120/134 to appId.GetAwaiter().GetResult(). That change makes the AggregateException block obsolete. I would suggest take what's in develop.